tests/cpp/test_multidevice_overlap.cpp

```diff
@@ -0,0 +1,283 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#ifdef NVFUSER_DISTRIBUTED
+#include <ATen/Functions.h>
+#include <c10/cuda/CUDAStream.h>
+#include <c10/util/ArrayRef.h>
+#include <fusion.h>
+#include <ir/utils.h>
+#include <ops/all_ops.h>
+#include <tests/cpp/multidevice.h>
+
+namespace nvfuser {
+
+int parseEnvVariable(const char* env_name) {
+  const std::string prefix = "NVFUSER_OVERLAP_";
+  auto prefixed_name = prefix + env_name;
+  auto env = std::getenv(prefixed_name.c_str());
+  if (!env) {
+    return -1;
+  }
+  return std::atoi(env);
+}
+
+bool isEnvVariableDefined(const char* env_name) {
+  return parseEnvVariable(env_name) != -1;
+}
+
+struct OverlapTestParams {
+  // Tensors sizes
+  int64_t M = std::pow(2, 6);
+  int64_t K = std::pow(2, 5);
+  int64_t N = std::pow(2, 4);
+  int64_t S = std::pow(2, 3);
+
+  // network backend type
+  CommunicatorBackend backend_type = CommunicatorBackend::nccl;
+
+  // overlap optimization parameters
+  bool use_different_streams =
+      false; // whether to change CUDA stream at each iteration
+
+  // debug
+  bool debug_print = false;
+
+  void parseEnv() {
+    if (isEnvVariableDefined("LOG2_M")) {
+      M = std::pow(2, parseEnvVariable("LOG2_M"));
+    }
+    if (isEnvVariableDefined("LOG2_K")) {
+      K = std::pow(2, parseEnvVariable("LOG2_K"));
+    }
+    if (isEnvVariableDefined("LOG2_N")) {
+      N = std::pow(2, parseEnvVariable("LOG2_N"));
+    }
+    if (isEnvVariableDefined("LOG2_S")) {
+      S = std::pow(2, parseEnvVariable("LOG2_S"));
+    }
+    if (isEnvVariableDefined("USE_UCC")) {
+      backend_type = CommunicatorBackend::ucc;
+    }
+    if (isEnvVariableDefined("USE_STREAMS")) {
+      use_different_streams = true;
```

Comment:
IIUC, "use different streams" is not just "use streams" but also use a different stream per iteration. So I'd reflect that by changing the flag to USE_DIFFERENT_STREAMS to be consistent.